strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticator.java

```diff
@@ -0,0 +1,29 @@
+package org.carlspring.strongbox.authentication.api.impl.ldap;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class LdapAuthenticator
```

Comment:
@sbespalov it's a leftover from my previous authentication mechanism assumptions where the authenticator contained both authentication supplier and authentication provider. Now those things are separated and I believe you're right we can get rid in the implementation terms.

However, `Authenticator` is a simple API for our security providers extension. See here for review:
https://github.com/strongbox/strongbox/wiki/How-To-Implement-A-Custom-Authenticator

Thanks to it, end-client will be able introduce authentication provider to the strongbox distribution just by adding a simple dependency to `<artifactId>strongbox-authentication-api</artifactId>`.  I need to sleep with it and think about it :) Please leave me you comments :)